IdentityCore/src/telemetry/request_telemetry/MSIDRequestTelemetryUtils.m

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+
+#import "MSIDRequestTelemetryUtils.h"
+#import "MSIDRequestTelemetryConstants.h"
+
+@implementation MSIDRequestTelemetryUtils
+
++ (nonnull NSString *)getStringFromBool:(BOOL)val
+{
+    if(val)
+    {
+        return MSID_REQUEST_TELEMETRY_SCHEMA_STRING_TRUE;
+    }
+    
+    return MSID_REQUEST_TELEMETRY_SCHEMA_STRING_FALSE;
+}
+
++ (BOOL)writeSignoutApplicationForTelemetry:(nonnull NSString *)applicationIdentifier
+                                      error:(NSError * _Nullable * _Nullable)error
+{
+    if (![applicationIdentifier isKindOfClass:[NSString class]])
+    {
+        NSString *errorDescription = @"Signout application identifier is nil or empty";
+        NSError *validationError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, errorDescription, nil, nil, nil, nil, nil, NO);
+        if (error)
+        {
+            *error = validationError;
+        }
+        return NO;
+    }
+    
+    [[NSUserDefaults standardUserDefaults] setObject:applicationIdentifier forKey:MSID_CURRENT_REQUEST_TELEMETRY_KEY_FLW_SIGNOUT_APP];
+    return YES;
+}
+
++ (nullable NSString *)readSignoutApplicationForTelemetry
+{
+   return [[NSUserDefaults standardUserDefaults] stringForKey:MSID_CURRENT_REQUEST_TELEMETRY_KEY_FLW_SIGNOUT_APP];
+}
+
++ (void)removeSignoutApplicationForTelemetry
```

Comment:
Do we need same for sign-in identifier?

Reply:
getting signin information from broker.request.sourceapplication